src/components/Map/YandexMapHelpers.ts

```diff
@@ -0,0 +1,169 @@
+import {Coordinate} from '../../models/constructor-items/common';
+import {loadScript} from '../../utils';
+import {Maplangs, MapType, MapsContextSourceLinks} from '../../context/mapsContext';
+
+import {YMapMarker} from '../../models';
+
+export interface IInitMapParams {
+    zoom?: number;
+    center?: Coordinate[];
+}
+
+const DEFAULT_PLACEMARKS_COLOR = '#dc534b';
+const DEFAULT_PLACEMARKS_PRESET = 'islands#dotIcon';
+const DEFAULT_ZOOM = 9;
+const DEFAULT_CENTER = [55.753994, 37.622093];
+
+export function initMap(containerId: string, initParams?: IInitMapParams) {
+    return new window.ymaps.Map(containerId, {
+        center: initParams?.center || DEFAULT_CENTER,
+        zoom: initParams?.zoom || DEFAULT_ZOOM,
+    });
+}
+
+export class YMapsApiLoader {
+    static isApiLoaded = false;
+    static loaders: Array<Promise<unknown>> = [];
+    static scriptSrc: string = MapsContextSourceLinks[MapType.Yandex];
+
+    static async loadApi(apiKey: string, lang: 'ru' | 'en', nonce?: string) {
+        if (YMapsApiLoader.loaders.length !== 0) {
```

Comment:
Didn't really understand - if we need to load y maps only one time why do we need this loaders query and if we need to load it many times simultaneously then why we use this static methods only class and not use just class instance for each load?